### PR TITLE
fixing default bundle version for logic apps

### LINKED ISSUE #7706: Default host.json generated for logic app is using the wrong verison
When creating a default host.json when a function app is created within the context of logic app it is using the wrong version. The default version for workflow bundle should be "[1.*,2.0.0).

### PR DESCRIPTION
**Pull request by soninaren**
closes #7706

### CODE REVIEW

File: src/WebJobs.Script.WebHost/host.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "version": "2.0",
+  "extensionBundle": {
+    "id": "Microsoft.Azure.Functions.ExtensionBundle",
+    "version": "[2.*, 3.0.0)"
+  }
+}
```
Comment (gzuber) [inline src/WebJobs.Script.WebHost/host.json]: Is this being used?

**Review by gzuber — CHANGES_REQUESTED** [review]
Looks good! Just one question

File: src/WebJobs.Script.WebHost/host.json (hunk 1 above)
Comment (soninaren) [reply src/WebJobs.Script.WebHost/host.json]: not sure where this came from. need to remove this.

**Review by gzuber — DISMISSED** [review]

File: src/WebJobs.Script/ScriptConstants.cs [hunk 2]
```diff
@@ -179,6 +179,8 @@ public static class ScriptConstants
         public const string Windows64BitRID = "win-x64";
         public const string Windows32BitRID = "win-x86";
         public const string WorkFlowAppKind = "workflowapp";
+        public const string LogicAppDefaultExtensionBundleVersion = "[1.*, 2.0.0)";
```
Comment (sidkri) [inline src/WebJobs.Script/ScriptConstants.cs:182]: It's not good practice to couple Functions host with Logic Apps this way.  Prefer if the user (Logic Apps infra) configured the bundle explicitly.

Comment (sidkri) [reply src/WebJobs.Script/ScriptConstants.cs:182]: discussed offline and understood that Logic Apps is only creating the Function app and not doing any code deployment so they can't configure the bundle version after app creation and we don't yet support configuring host.json at app creation.  

Could you please add comments around this explaining that this is temporary until we support configuring host.json at app creation time (via ARM template / REST API at minimum)

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs [hunk 3]
```diff
@@ -215,7 +215,11 @@ internal JObject LoadHostConfig(string configFilePath)
                                             ScriptConstants.WorkFlowExtensionBundleId :
                                             ScriptConstants.DefaultExtensionBundleId;
 
-                        hostConfigObject = TryAddBundleConfiguration(hostConfigObject, bundleId);
+                        string bundleVersion = _configurationSource.Environment.IsLogicApp() ?
```
Comment (sidkri) [inline src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs:221]: Could you please add comments around this explaining that this is temporary until we support configuring host.json at app creation time (via ARM template / REST API at minimum)

File: test/WebJobs.Script.Tests/Configuration/HostJsonFileConfigurationSourceTests.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Azure.WebJobs.Script.Tests.Configuration
     public class HostJsonFileConfigurationSourceTests
     {
         private readonly string _hostJsonWithBundles = "{\r\n  \"version\": \"2.0\",\r\n  \"extensionBundle\": {\r\n    \"id\": \"Microsoft.Azure.Functions.ExtensionBundle\",\r\n    \"version\": \"[2.*, 3.0.0)\"\r\n  }\r\n}";
-        private readonly string _hostJsonWithWorkFlowBundle = "{\r\n  \"version\": \"2.0\",\r\n  \"extensionBundle\": {\r\n    \"id\": \"Microsoft.Azure.Functions.ExtensionBundle.Workflows\",\r\n    \"version\": \"[2.*, 3.0.0)\"\r\n  }\r\n}";
+        private readonly string _hostJsonWithWorkFlowBundle = "{\r\n  \"version\": \"2.0\",\r\n  \"extensionBundle\": {\r\n    \"id\": \"Microsoft.Azure.Functions.ExtensionBundle.Workflows\",\r\n    \"version\": \"[1.*, 2.0.0)\"\r\n  }\r\n}";
```
Comment (sidkri) [inline test/WebJobs.Script.Tests/Configuration/HostJsonFileConfigurationSourceTests.cs:23]: I don't see this test changing to run as LogicApp so why did you change the default to the older version?

**Review by sidkri — CHANGES_REQUESTED** [review]
Please address the comments and I'll approve after another quick review.

File: test/WebJobs.Script.Tests/Configuration/HostJsonFileConfigurationSourceTests.cs (hunk 4 above)
Comment (soninaren) [reply test/WebJobs.Script.Tests/Configuration/HostJsonFileConfigurationSourceTests.cs:23]: Not sure what you mean by that

Comment (soninaren) [reply test/WebJobs.Script.Tests/Configuration/HostJsonFileConfigurationSourceTests.cs:23]: The test was already running as a logic app. I had changed this in error in the previous change.

**Review by sidkri — APPROVED** [review]
